apps/game_backend/lib/game_backend/users.ex

```diff
@@ -113,4 +130,56 @@ defmodule GameBackend.Users do
     })
     |> Repo.update()
   end
+
+  defp apply_currency_rewards(user_id, currency_rewards) do
+    Enum.map(currency_rewards, fn currency_reward ->
+      Currencies.add_currency(user_id, currency_reward.currency_id, currency_reward.amount)
+    end)
+    |> check_result(:currency_rewards)
+  end
+
+  defp build_item_rewards_params(user_id, item_rewards) do
+    Enum.map(item_rewards, fn item_reward ->
+      Enum.map(1..item_reward.amount, fn _ ->
+        %{
+          user_id: user_id,
+          template_id: item_reward.item_template_id,
+          level: 1,
+          inserted_at: NaiveDateTime.utc_now() |> NaiveDateTime.truncate(:second),
+          updated_at: NaiveDateTime.utc_now() |> NaiveDateTime.truncate(:second)
+        }
+      end)
+    end)
+    |> List.flatten()
+  end
+
+  defp build_unit_rewards_params(user_id, unit_rewards) do
+    IO.inspect(unit_rewards, label: :rewards)
```

Comment:
lets delete this

Reply:
Whoops, good thing we have credo ci now :P